homeassistant/components/ambient_station/binary_sensor.py

```diff
@@ -62,11 +191,11 @@ class AmbientWeatherBinarySensor(AmbientWeatherEntity, BinarySensorEntity):
     @callback
     def update_from_latest_data(self) -> None:
         """Fetch new state data for the entity."""
-        state = self._ambient.stations[self._mac_address][ATTR_LAST_DATA].get(
-            self._sensor_type
-        )
+        state = self._ambient.stations[self._mac_address][ATTR_LAST_DATA][
+            self.entity_description.key
+        ]
 
-        if self._sensor_type in (
+        if self.entity_description.key in (
```

Comment:
Try using an in-place `set` instead of `tuple`. They are more efficient when it comes to large lookups.
At runtime the Python interpreter will convert it to a `frozenset` automatically.

Reply:
Just though of something else. Why not use a custom field in the description for it. You could define a lambda for `is_on` and just call it here. Then you wouldn't need the lookup at all.

Reply:
I don't like the idea of mixing state calculation inside of an `EntityDescription`. If we set that standard here, why wouldn't every entity use a "description lambda" to calculate state?

Reply:
Most integrations are just too different for it to make sense in general. (At least not at the moment.)

However, there are a few were it does work quite well.
https://github.com/home-assistant/core/blob/6a93f5b7ad1fc751be8ac3a57aea10f9858a7b6e/homeassistant/components/melcloud/sensor.py#L160-L162
https://github.com/home-assistant/core/blob/6a93f5b7ad1fc751be8ac3a57aea10f9858a7b6e/homeassistant/components/fritzbox/binary_sensor.py#L50-L52

Or a bit different
https://github.com/home-assistant/core/blob/6a93f5b7ad1fc751be8ac3a57aea10f9858a7b6e/homeassistant/components/renault/binary_sensor.py#L66-L71

Reply:
You wouldn't even need to store a lambda. It should be enough to store if `high` (1) or `low` (0) should be `on`.

Reply:
Got it – helpful! 👍🏻